### PR TITLE
CORTX-23733 - Improved links in developers guide (motr-developer-guide.md)

### PR DESCRIPTION
**Pull request by Garuda5611**
Signed-off-by: Pratik Patil <pratik.k.patil@seagate.com>

# Problem Statement
- Improved links in developers guide

# Design
-  Links present in current document were just references, so added complete URL to this files.
- Added links to other sections in the same document.

# Coding
   Checklist for Author
-  [ ] Coding conventions are followed and code is consistent

# Testing 
  Checklist for Author
- [ ] Unit and System Tests are added
- [ ] Test Cases cover Happy Path, Non-Happy Path and Scalability
- [ ] Testing was performed with RPM

# Impact Analysis
  Checklist for Author/Reviewer/GateKeeper
- [ ] Interface change (if any) are documented
- [ ] Side effects on other features (deployment/upgrade)
- [ ] Dependencies on other component(s)

# Review Checklist 
  Checklist for Author
- [ ] JIRA number/GitHub Issue added to PR
- [ ] PR is self reviewed
- [ ] Jira and state/status is updated and JIRA is updated with PR link
- [ ] Check if the description is clear and explained

# Documentation
  Checklist for Author
- [ ] Changes done to WIKI / Confluence page / Quick Start Guide

-----
[View rendered doc/motr-developer-guide.md](https://github.com/Garuda5611/cortx-motr/blob/CORTX-23733/doc/motr-developer-guide.md)

### CODE REVIEW

**Comment by rkothiya** [comment]
Can one of the admins verify this patch?

File: doc/motr-developer-guide.md [hunk 1]
```diff
@@ -4,35 +4,37 @@ This document describes how to develop a simple application with Motr APIs.
...
 
 For your convenience to build and develop motr, we offer a
 [CentOS-7.9-based Development VM](https://github.com/Seagate/cortx-motr/releases/tag/ova-centos79)
 with all dependencies installed ready to build and start single-node Motr cluster.
 
 CORTX Motr provides object based operations and index (a.k.a key/value) based operations as well as FDMI interface.
-For architectural documents, please refer to our [reading list](reading-list.md).
+For architectural documents, please refer to our [reading list](https://github.com/Seagate/cortx-motr/blob/main/doc/reading-list.md).
 
 ## A simple CORTX Motr Object Application
```
Comment (rkothiya) [inline doc/motr-developer-guide.md:25]: This is being shown as a broken link on rendering.

File: doc/motr-developer-guide.md [hunk 2]
```diff
@@ -4,35 +4,37 @@ This document describes how to develop a simple application with Motr APIs.
...
 
-[Object Application](motr-object-app.md)
++  [A simple Cortx Motr application (object)](https://github.com/Seagate/cortx-motr/blob/main/doc/motr-object-app.md)
++  [A simple Cortx Motr application (index)](#a-simple-cortx-motr-index-key-value-application)
```
Comment (rkothiya) [inline doc/motr-developer-guide.md:28]: This and the below link are pointing to the same place

**Review by rkothiya — APPROVED** [review]
LGTM

**Review by anaghadeshmukh — APPROVED** [review]

**Comment by madhavemuri** [comment]
I can see the links properly in main as well, https://github.com/Seagate/cortx-motr/blob/main/doc/motr-developer-guide.md
I didn't get why these are needed.

**Review by huanghua78 — APPROVED** [review]

**Comment by rkothiya** [comment]
retest this please

**Comment by huanghua78** [comment]
> I can see the links properly in main as well, https://github.com/Seagate/cortx-motr/blob/main/doc/motr-developer-guide.md I didn't get why these are needed.

I don't see why this is needed either.

**Pull request closed without merging**